setup.py

```diff
@@ -106,7 +106,12 @@ def get_generator_flags():
     elif is_win:
         flags.append("\"Visual Studio 9 2008\"" if is_27 else "\"Visual Studio 15 2017\"")
         flags.append("-A")
-        flags.append("x64" if is_x64 else "Win32")
+        target_map = {"AMD64": "x64",
+                      "IA64" : "x64",
+                      "ARM64": "ARM64",
+                      "x86"  : "Win32"}
+        target = target_map[platform.machine()]
```

Comment:
This would fail if machine is not one of the 4 keys. For instance, the doc says that if the machine is unknown, an empty string is returned. Also, we have no guarantee that those 4 keys are the only possible one.
I would introduce a fallback to "Win32", which may be weak but better than a KeyError (and consistent with current behavior).

https://docs.python.org/3/library/platform.html#platform.machine

Reply:
Thinking more, a more targetting approach might be something like:
```
flags.append("ARM64" if platform.machine() == "ARM64" else "x64" if is_x64 else "Win32")
```

Reply:
Or if you can't handle the platform detection, just leave the flag off and let CMake do its own thing?

Reply:
Thanks for the review. Happy to go with any suggestion.

I think following might be bit more future proof.

```
target_map = {"AMD64": "x64",
                         "IA64" : "x64",
                          "ARM64": "ARM64",
                          "x86"  : "Win32"}

 if platform.machine() in target_map:
            flags.append("-A {}".format(target_map[platform.machine()]))
```

Let me know what you guys prefer.

Reply:
hey @nsait-linaro,

for the sake of consistent compiling behavior with previous versions (either compilation success or failure behavior), I prefer the targeted approach suggested by Laurent which to me is more of an incremental change and is safer to address the ARM64 compiler issue:

> flags.append("ARM64" if platform.machine() == "ARM64" else "x64" if is_x64 else "Win32")

Reply:
Yes sure, I will update the patch

Reply:
Patch updated with `flags.append("ARM64" if platform.machine() == "ARM64" else "x64" if is_x64 else "Win32")`

Reply:
thanks!

Reply:
Yes I am using native arm64 python on an arm64 windows machine. Please let me know how can I be of help. 

(Sorry I missed your message earlier)